ddl/index_merge_tmp.go

```diff
@@ -48,25 +48,12 @@ func (w *mergeIndexWorker) batchCheckTemporaryUniqueKey(txn kv.Transaction, idxR
 		return errors.Trace(err)
 	}
 
-	// 1. unique-key/primary-key is duplicate and the handle is equal, skip it.
-	// 2. unique-key/primary-key is duplicate and the handle is not equal, return duplicate error.
-	// 3. non-unique-key is duplicate, skip it.
 	for i, key := range w.originIdxKeys {
 		if val, found := batchVals[string(key)]; found {
-			if idxRecords[i].distinct && !bytes.Equal(val, idxRecords[i].vals) {
-				return kv.ErrKeyExists
-			}
-			if !idxRecords[i].delete {
-				idxRecords[i].skip = true
-			} else {
-				// Prevent deleting an unexpected index KV.
-				hdInVal, err := tablecodec.DecodeHandleInUniqueIndexValue(val, w.table.Meta().IsCommonHandle)
-				if err != nil {
-					return errors.Trace(err)
-				}
-				if !idxRecords[i].handle.Equal(hdInVal) {
-					idxRecords[i].skip = true
-				}
+			// Found a value in the original index key.
+			err := checkTempIndexKey(txn, idxRecords[i], val, w.table)
```

Comment:
if the err is not duplicate key, what react about add index tasks? will it rollback? or just retry?